src/components/SelectedSurveyPage.jsx

```diff
@@ -1,36 +1,105 @@
 import React, { PropTypes } from 'react';
 import PureComponent from 'react-pure-render/component';
+import Immutable from 'immutable';
+
+import SurveyQuestion from 'components/SurveyQuestion';
+import { MARGIN_BETWEEN_QUESTIONS, SELECTED_SURVEY_BODY_PADDING } from 'constants';
 
 export default class SelectedSurveyPage extends PureComponent {
   static propTypes = {
     id: PropTypes.number.isRequired,
-    survey: PropTypes.object.isRequired,
+    survey: PropTypes.instanceOf(Immutable.Map).isRequired,
     fetchSurvey: PropTypes.func.isRequired,
     saveSurvey: PropTypes.func.isRequired,
-  }
+  };
 
   static contextTypes = {
     router: React.PropTypes.object.isRequired,
   };
 
-  componentDidMount() {
+  state = {
```

Comment:
Wait, what's wrong with the style here? Initializing the state is 1) better than using `this.state` without initializing it and 2) more cleaner than doing this in the constructor, even though it gets executed in the constructor (see https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#es7-property-initializers and https://babeljs.io/blog/2015/06/07/react-on-es6-plus).

And if by bad style, you mean that it we shouldn't use `state` at all, then (although I agree that would be ideal) we can't do that. We need to use the state here because this information cannot be derived from the `props` and putting it inside the `redux` store would just pollute it.